test/tests/functional/pbs_test_tpp.py

```diff
@@ -110,38 +228,95 @@ def test_comm_with_mom(self):
-                      ATTR_q: resv_que,
-                      ATTR_l + '.place': 'scatter'}
-        jid = self.submit_job(set_attrib, job_script=True)
-        # Wait for reservation to start
-        a = {'reserve_state': (MATCH_RE, 'RESV_RUNNING|5')}
-        self.server.expect(RESV, a, rid, offset=10)
-        self.server.expect(JOB, {'job_state': 'R'}, id=jid)
+        self.common_steps(job=True, interactive_job=True, resv=True)
+
+    @requirements(num_moms=2, num_comms=1, num_clients=1)
+    def test_client_with_mom(self):
+        """
+        This test verifies communication between server-mom,
+        server-client and between moms through pbs_comm
+        Configuration:
+        Node 1 : Server, Mom, Sched, Comm
+        Node 2 : Mom
+        Node 3 : Client
+        """
+        msg = "Need atleast 2 moms and 1 client as input. "
+        msg += "use -pmoms=<m1>:<m2>,client=<c1>"
+        if len(self.moms) < 2 and self.server.client == self.server.hostname:
+            self.skip_test(reason=msg)
+        self.momA = self.moms.values()[0]
+        self.momB = self.moms.values()[1]
+        self.hostA = self.momA.shortname
+        self.hostB = self.momB.shortname
+        self.hostC = self.server.client
+        nodes = [self.hostA, self.hostB, self.hostC]
+        self.node_list.extend(nodes)
+        self.server.manager(MGR_CMD_SET, SERVER, {'flatuid': True})
+        self.common_steps(job=True, interactive_job=True)
+        self.server.client = self.hostB
+        self.common_steps(resv=True)
+
+    @requirements(num_moms=2, num_comms=1, num_clients=1)
+    def test_comm_non_server_host(self):
+        """
+        This test verifies communication between server-mom,
+        server-client and between moms through pbs_comm which
+        is running on non server host
+        Configuration:
+        Node 1 : Server, Mom, Sched
+        Node 2 : Client
+        Node 3 : Mom
+        Node 4 : Comm
+        """
+        msg = "Need atleast 2 moms and a comm installed on host"
+        msg += " other than server host"
+        if len(self.moms) < 2 or len(self.comms) < 1:
+            self.skip_test(reason=msg)
+        self.momA = self.moms.values()[0]
+        self.momB = self.moms.values()[1]
+        self.comm1 = self.comms.values()[0]
+        self.hostA = self.momA.shortname
+        self.hostB = self.server.client
+        self.hostC = self.momB.shortname
+        self.hostD = self.comm1.shortname
+        nodes = [self.hostA, self.hostB, self.hostC, self.hostD]
+        self.node_list.extend(nodes)
+        a = {'PBS_START_COMM': '0', 'PBS_START_MOM': '1',
+             'PBS_LEAF_ROUTERS': self.hostD}
+        self.set_pbs_conf(host_name=self.server.hostname, conf_param=a)
+        a = {'PBS_LEAF_ROUTERS': self.hostD}
+        self.set_pbs_conf(host_name=self.hostC, conf_param=a)
+        self.common_steps(job=True, resv=True)
+        self.server.client = self.hostB
```

Comment:
same comment as before - pass client=self.hostB to common_steps() and then to submit_job().